cluster/log-dump/log-dump.sh

```diff
@@ -141,7 +141,7 @@ function save-logs() {
       fi
     else
       case "${KUBERNETES_PROVIDER}" in
-        gce|gke)
+        gce|gke|kubernetes-anywhere)
```

Comment:
kube-anywhere as a provider?  what?  Most of these values are to detect cloud provider changes.  I srsly do not want to crib kube-anywhere as a provider.

Reply:
It's been there in the current CI jobs for a while

https://github.com/kubernetes/test-infra/blob/master/images/kubeadm/Dockerfile#L28

Also see a recent run, look for `--provider=kubernetes-anywhere` and `KUBERNETES_PROVIDER=kubernetes-anywhere`

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/54584/pull-kubernetes-e2e-kubeadm-gce/8640/build-log.txt

Not introducing anything new here. just surfacing it. The "normal" GCE jobs have a different names for the nodes vs the kubernetes-anywhere/GCE jobs